test/e2e/specs/fakerp/metricsbridge.go

```diff
@@ -0,0 +1,114 @@
+package fakerp
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"regexp"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/ghodss/yaml"
+	"github.com/prometheus/client_golang/api"
+	"github.com/prometheus/client_golang/api/prometheus/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/openshift-azure/pkg/entrypoint/metricsbridge"
+	"github.com/openshift/openshift-azure/pkg/util/roundtrippers"
+	"github.com/openshift/openshift-azure/test/sanity"
+)
+
+// Test checks if all configured queries are valid prometheus queries, and if the metric exists
+// Existence of metric is checked by either:
+//  - query returning data
+//  - query being on a list of manually verified correct queries (that normally don't return any data)
+//  - query having ' != 0' at the end, that returns data after this condition is removed
+//  - query calculating rate( x [1m]) of a metric x, when querying for metric x returns data
+var _ = Describe("Metricsbridge E2E check configured queries [Fake]", func() {
+	// queries which usually return no datapoints, but have been verified as valid manually
+	manuallyVerifiedQueries := map[string]bool{
```

Comment:
I am wondering if the following will get rid of the need for manuallyVerifiedQueries:
1) run all the queries by prometheus (should catch syntax errors)
2) fetch and parse the output of the federate endpoint (this lists all known metrics); for each query extract the metric name and ensure it's in the federate list
(for a hint on fetching and parsing the federate endpoint see an older version of metricsbridge which had this code)